pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -255,6 +258,21 @@ stdenv.mkDerivation (finalAttrs: {
     ./0019-meson-Don-t-link-ssh-dropins.patch
 
     ./0020-install-unit_file_exists_full-follow-symlinks.patch
+
+    # add nspawn build option flag
+    # required to disable nspawn for systemdLibs to avoid dependency on getent
+    # https://github.com/systemd/systemd/pull/36876, remove for systemd 258
+    (fetchpatch {
+      # required for the actual patch to apply
+      url = "https://github.com/systemd/systemd/commit/b1fb2d971c810e0bdf9ff0ae567a1c6c230e4e5d.patch";
+      hash = "sha256-JBheazg1OFkx8vUl2l8+34BoEPVURBQJHxqntOBYB60=";
+      includes = [ "src/nspawn/meson.build" ];
+    })
+    (fetchpatch {
+      url = "https://github.com/systemd/systemd/commit/d95818f5221d9b9b19648cffa0cb2407f023b27e.patch";
+      hash = "sha256-FTpWGec5ivlkyEEDMCPaLE+BH91e7JI0kH8pS88bBDY=";
+      excludes = [ "test/fuzz/meson.build" ];
+    })
```

Comment:
TODO: wait for https://github.com/NixOS/nixpkgs/pull/427968

Reply:
Maybe we just disable the logind support for now in pam on musl on since that wasn't present before either and re-enable it once we have the new systemd version.

Reply:
utmp has to die, and if logind is the replacement, we should enable it. pam was/is unmaintained in nixpkgs, it was long overdue someone actually took a look at it. As it stands, this PR is a full "lets do it properly" PR.
I also don't see the issue with patching in the nspawn toggle: our `systemdLibs` does not *install* nspawn anyways, so we might as well not build it either. This does not actually change anything in the package output, so i am not sure what the fuss is about.